src/hooks/useApiService.ts

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect, useRef } from 'react';
+import { OperationStatus } from 'services/api/dtos';
+
+export interface APIServiceResponse<E>{
+  data: E | undefined;
+  loading: boolean;
+  error: string | undefined;
+} 
+
+
+export type IApiMethod<E> = (args: Record<string, string | number>) => Promise<OperationStatus<E>>
+export type IUseApiService = <E>(method: IApiMethod<E>) => (args: Record<string, string | number>) => APIServiceResponse<E>
+
+export function useApiService(apiService: unknown): IUseApiService {
```

Comment:
I would suggest another name, something along the lines of useHttpMethod or useFetcher